build/gulpfile.extensions.js

```diff
@@ -336,4 +345,47 @@ async function webpackExtensions(taskName, isWatch, webpackConfigLocations) {
 	});
 }
 
+/**
+ * @param {string} taskName
+ * @param {boolean} isWatch
+ * @param {string | undefined} outputRoot
+ * @param {string} esbuildScripts
+ */
+async function esbuildExtensions(taskName, isWatch, outputRoot, esbuildScripts) {
+	function reporter(/** @type {string} */ stdError, /** @type {string} */script) {
+		const matches = (stdError || '').match(/\> (.+): error: (.+)?/g);
+		fancyLog(`Finished ${ansiColors.green(taskName)} ${script} with ${matches ? matches.length : 0} errors.`);
+		for (const match of matches || []) {
+			fancyLog.error(match);
+		}
+	}
+
+	const scripts = esbuildScripts.map(script => {
```

Comment:
esbuild has a pretty good JavaScript API they they actually recommend using for more complicated builds. Could consider that to reduce complexity and child process overhead here.

Reply:
Oh, I see you use that below. I think `esbuild.build` will spawn a child process with the native esbuild binary, so nesting a child process may be unnecessary